controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -76,6 +74,38 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// - ocmApiUrl: Optional, defaults to 'https://api.openshift.com'
 	// +optional
 	CredentialsSecretRef *corev1.LocalObjectReference `json:"credentialsSecretRef,omitempty"`
+
+	// Network config for the ROSA-HCP cluster.
+	// +optional
```

Comment:
globally: any commands for generators in comments should be their own comment above, so the content of the tag is not erroneously added to the GoDoc. For instance:


```go
	// +optional

	// Network config for the ROSA-HCP cluster.
	Network *NetworkSpec `json:"network,omitempty"`
```

Having a separate comment block above is the preferred way to do this.

Reply:
These markers are used to generate explicit fields in the OpenAPI and structural schema, which mark the fields as optional, etc, and are used to drive the UIs in the OpenAPI Explorer, etc. This is much nicer than just happening to put a comment in the doc - if you want to also add an English phrase for this, you are free to, but the inventors of all of these tags very explicitly made them work in the comment block above for this use-case. Review the Kubernetes API types and you'll see what I mean.

Reply:
I feel its not really clear what you mean @stevekuznetsov , Usually the markers tags under the comments. Do we miss understand you ? would you give reference to what you mean by example.

Reply:
I left an example in my [comment](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4780#discussion_r1481839594) but please feel free to also look at any of the Kubernetes API definitions for other examples, or even in this repo where the guidance is being followed for types like [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/6ffb575b3f78a6affbc792c0f7902cb8b84cc77e/exp/api/v1beta2/awsfargateprofile_types.go#L125-L133) but not for their fields.

Reply:
I haven't seen anyone putting the `+optional` marker above, not in this repo and not in k8s apis, example: https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L118

Reply:
It's best practice and why the libraries support it. What's the downside? Why all this pushback? It's not good to put machine-readable comments into user-facing documentation. Not `// +optional` and not `// +listType=map`.

Reply:
Would be good to articulate what either of you see as a downside here.

Reply:
Its not about downsides or pushback, I just see a consistent style used across the repo and the core k8s apis. My question is why would we go out of our way to change it?
is there a reference for the best practices?

Reply:
"Going out of our way" == maybe ten seconds of work so I don't really buy that ;)

The docs [hilariously](https://book.kubebuilder.io/cronjob-tutorial/api-design) expect that you follow this pattern for kube-builder flags but not `// +optional.` The logic here is really, really, simple - these are machine-readable patterns that should not be in human-readable documentation. That is exactly why the libraries allow you to put the flags in the comment block above, so they don't end up in GoDoc or OpenAPI etc. From my recollection, this was figured out after the first APIs were written, so for backwards compatibility it's allowed to have them in the main GoDoc comment. 

Humans - when they aren't the devs writing the code - always prefer to read real prose with real meaning, rather than machine-readable tags.

I don't know that it's super useful to talk about what the "consistent style" is because even in this repo the guidance is followed haphazardly and (IMO) without much thought. For instance, the tags are (correctly) split out for `struct`-level comments, but are not for fields. If I had to hazard a guess as to why, it's just copy-pasta that stuck around and nobody cared. Since this PR was changing API contracts, it seemed beyond trivial and simple to ask Mohamed to edit a GoDoc comment. Does it leave the struct in a half-state? Sure. We can follow up with a PR to bring everything inline. There's value in updating things as you work on them and there's some value in consistency in the repo. I'd rather us just move on as at this point we've invested an incredibly large amount of time talking back and forth.